cpp/src/arrow/compute/exec/bloom_filter_test.cc

```diff
@@ -309,7 +309,7 @@ Status TestBloomSmall(BloomFilterBuildStrategy strategy, int64_t num_build,
 
   int64_t num_repeats = 1LL;
 #ifdef NDEBUG
-  num_repeats = std::max(1LL, bit_util::CeilDiv(1000000ULL, num_probe));
+  num_repeats = std::max(static_cast<int64_t>(1), bit_util::CeilDiv(1000000ULL, num_probe));
```

Comment:
It's possible I don't understand but is `static_cast<int64_t>(1)` any different than `1LL`?

Reply:
I got a whole bunch of errors in release mode with`1LL` because I guess sometimes `int64_t` is a `long` and sometimes a `long long`. I think it's the same reason we have `int64_for_gather_t`.